### PR TITLE
Parser Fixes & The great renaming! f=kewisch

### PR DESCRIPTION
**Pull request by lightsofapollo**
Correctly upgrades date formats for period and utc-offset.

Also the renaming changes we have been talking about.

The PR is not ready quite yet I am going to try to port the recurring value parser to a simpler solution (while keeping most/all of the validation) then we can remove decoration_parser and we should be around the same sloc (or lower) then before the parser changes.

### CODE REVIEW

**Comment by lightsofapollo** [comment]
@kewisch You can take a look now if you like. Tests fail ( I broke recurring event fromString) but that is the last piece the rest will stay as is unless I find a bug somewhere.

File: lib/ical/utc_offset.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      return (this.factor == 1 ? "+" : "-") +
+              ICAL.helpers.pad2(this.hours) + ':' +
+              ICAL.helpers.pad2(this.minutes);
+    }
+  };
+
+  UtcOffset.fromString = function(aString) {
+    // -05:00
+    var options = {};
+    //TODO: support seconds per rfc5545 ?
```
Comment (kewisch) [inline lib/ical/utc_offset.js:27]: I believe you support seconds somewhere else, at least I remember reading something around those lines. Its probably not too much extra, so go ahead and add seconds support.

**Comment by kewisch** [comment]
r=philipp for these bits, of course I'd appreciate if you could make the tests pass before merging.

File: lib/ical/utc_offset.js (hunk 1 above)
Comment (lightsofapollo) [inline lib/ical/utc_offset.js:27]: correct, I will add support here now.

**Comment by lightsofapollo** [comment]
@kewisch Ready for final review. We are about ~100 less sloc then the pre1 build (prior to all these "v2" changes).

**Comment by lightsofapollo** [comment]
@kewisch Important to note all "decorated" objects return their jCal string value _not_ their ICAL (rfc 5455) value.
Support to go from one to the other is easy but I added a convince toICALString to ICAL.Time. Maybe I should do the same for the other decorated classes?

**Comment by lightsofapollo** [comment]
@kewisch Easy meaning you can always do something like this:

``` js
ICAL.design.value['my-type'].toICAL(jCalStringValue);
```

**Comment by kewisch** [comment]
Yes, I think convenience methods would be nice here. r=philipp

**Comment by lightsofapollo** [comment]
@kewisch Landing as is and will take the toICALString methods as a follow up today (those jCal string changes where introduced before this PR).

**Pull request merged**